ui/dashboards/src/components/EmptyDashboard/EmptyDashboard.tsx

```diff
@@ -35,50 +37,85 @@ export interface EmptyDashboardProps {
   /**
    * Components that are placed below the title and description that include
    * actions for the user to take (e.g. buttons or links). If not specified,
-   * the default "add panel" and "add variable" buttons will be displayed. Set
-   * to `false` to disable the default buttons.
+   * the default  buttons will be displayed. Set to `false` to disable the default
+   * buttons.
    */
-  actions?: boolean | React.ReactNode;
+  actions?: JSX.Element | boolean;
+
+  /**
+   * Handler for clicking the edit button when the dashboard is in "view" mode.
+   * Required when using the default empty state.
+   */
+  onEditButtonClick?: () => void;
 }
```

Comment:
I originally tried to make a discrimination union for the props to assist with the custom vs. default use cases, but it caused a LOT of headaches with TS and didn't seem to work very well, so I scrapped it and kept it simpler like it is now.